src/bt_peer_connection.cpp

```diff
@@ -414,13 +414,13 @@ namespace {
 
 		if (!m_supports_fast) return;
 
-		std::shared_ptr<torrent> t = associated_torrent().lock();
```

Comment:
I don't think this is correct, because the asserts are not related to the logging. Maybe what do you need here is a `#if TORRENT_USE_ASSERTS` block

Reply:
![default](https://user-images.githubusercontent.com/920705/34651946-d954d1bc-f3e8-11e7-9b33-2f16a2fb27d5.png)

?